ls1intum/Artemis
# `Discussion`: Improve design of posting headers and footers

### Checklist
#### General


#### Client

### Motivation and Context
This PR aims to enhance the design of the Post and AnswerPost components since suggestions were made about the current style.

### Description

- To make the design more consistent, we move the emoji picker for Posts to the left-hand side of the Pin and Archive buttons. This way, the picker is positioned right after the reactions users attach to Posts, the way it is already done for AnswerPosts. (instructor account needed to observe the Pin & Archive buttons)
- We integrate the "Answer Now" button to the emoji bar of the last AnswerPost of a Post if it was already answered. This way we reduce the height the Post and its' AnswerPosts take, reducing the scrolling users perform to reach information.
- We redesign the headers of Posts and AnswerPosts to stress the author more and make it easier to read together with the Date/Time a Posting was created.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Lecture with Discussions enabled

1. Log in to Artemis as Instructor
2. Navigate to Course Discussions
3. Verify that the emoji picker is on the left-hand side of the pin and archive buttons.
4. Verify that the "Reply Now" button is integrated to the emoji-bar of the last Answer of a Post.
5. Verify that the Post and AnswerPost's author name is more clearly visible than the previous versions.
6. Select an answer to a post as "resolves post" by checking the checkmark at the top-right corner.
7. Verify that Answers are reordered accordingly. (Answers that resolve should appear at the very top, the rest should be ordered according to creation date)
8. Log in as a student.
9. React to a Post Answer with multiple answers.
10. Make sure nothing unexpected happens.
11. While keeping the instructor tab open, send/edit an answer as a student and verify that the WebSocket subscription still works correctly and AnswerPosts are ordered once again by observing the page from the instructor's tab. (no refreshing of the page is needed to see incoming/updated answers.)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
Same as before

### Screenshots


![image](https://user-images.githubusercontent.com/67344158/169657178-9397a54e-fdf2-4c4f-a8a9-5fe51e433552.png)
![image](https://user-images.githubusercontent.com/67344158/169657195-8e84a6b3-bdd4-4f82-9629-1a6d8481e1fb.png)
